IM_calculation/IM/read_waveform.py

```diff
@@ -168,8 +168,9 @@ def read_ascii_folder(path, station_names, comp, units="g"):
                 c: open(os.path.join(path, f"{station}.{c.str_value}")) for c in comp
             }
         except IOError:
+            files_names = f"{station}.{{{[c.str_value for c in comp]}}}"
```

Comment:
Nitpicking here, but given that this variable is only used once. I suggest to either:
a) not define the variable and put the logic in the string formatting OR
b) define a variable, but put all the logic in its definition (i.e. currently the join is done in the string, and the rest is done in the variable definition), e.g. something like
```python
file_paths = os.path.join(path, f"{station}.{{{[c.str_value for c in comp]}}})
```

Reply:
Ah yeah, option b makes sense.
Option a includes more nesting " marks than I want to have to think about.